File: PULSE.Model/OrderDetail.cs
namespace PULSE.Model
{
    public class OrderDetail : BaseModel
    {
        public int Id { get; set; }
        public int? OrderId { get; set; }
        public int ProductId { get; set; }
        public int? BicycleSizeId { get; set; }
        public decimal? UnitPrice { get; set; }
        public int? Quantity { get; set; }
        public string? Discriminator { get; set; }

        public virtual BicycleSize? BicycleSize { get; set; }
        //public virtual OrderHeader? Order { get; set; }
        public virtual ProductAIO? Product { get; set; }
    }
}

File: pulse_mobile/lib/model/orders/order_detail.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pulse_mobile/model/abstract/base_model.dart';
import 'package:pulse_mobile/model/abstract/product.dart';
import 'package:pulse_mobile/model/bicycle_size/bicycle_size.dart';

part 'order_detail.g.dart';

@JsonSerializable()
class OrderDetail extends BaseModel {
  int? productId;
  int? quantity;
  int? bicycleSizeId;
  double? unitPrice;
  String? discriminator;
  BicycleSize? bicycleSize;
  Product? product;

  OrderDetail();

  factory OrderDetail.fromJson(Map<String, dynamic> json) =>
      _$OrderDetailFromJson(json);

  Map<String, dynamic> toJson() => _$OrderDetailToJson(this);
}
